code/model/DNEnvironment.php

```diff
@@ -654,10 +654,12 @@ protected function getCommitData($sha) {
 	}
 
 	/**
+	 * @param string $action
+	 *
 	 * @return string
 	 */
-	public function Link() {
-		return $this->Project()->Link() . "/environment/" . $this->Name;
+	public function Link($action = '') {
+		return \Controller::join_links($this->Project()->Link(), "environment", $this->Name, $action);
```

Comment:
This is not strictly related to the PR, but I thought I would clean it up while I was around the neighbourhood